best-practices-in-kubernetes-security.md

```diff
@@ -0,0 +1,180 @@
+
+This would help you to avoid potential downtimes when deploying mission-critical applications. These controls exist as policies to enable you to restrict access to resources in the cluster.
+
+### **Monitor Network Traffic**
+
+Ideally, you should keep an eye on network activity and compare it to authorized traffic and traffic permitted by the Kubernetes network policy. The goal is to eliminate connections that are no longer required in order to minimize the attack surface. Additionally, you may strengthen the permitted network policy by removing any connections that aren't absolutely required.
+
+### **Adopt a Multi-Layered Security Approach**
+
+Cloud-native applications have diverse components and elements, making it imperative to look beyond containerized workloads to safeguard your Kubernetes deployments. Hence to secure your Kubernetes environment, you should adopt a multi-layered security strategy. The Kubernetes documentation recommends a multi-layered approach called 4C (Cloud, Container, Cluster and Cloud) for securing your Kubernetes environments.
+
+-   Cloud and data center infrastructure and other resources required to create Kubernetes environments
+
+-   Container clusters that comprise the workloads that run on a particular cluster
+
+-   Individual Containers, i.e., the software images used to instantiate workloads.
+
+-   Source Code and container images
+
+It would help if you took advantage of recommended tools to secure and protect each of these layers. However, you should know that you don't have one tool that can help protect all these layers.
+
+### **Enable Kubernetes Role-Based Access Control**
+
+Role-Based Access Control or RBAC is a critical security component of Kubernetes and one of the Kubernetes Deployment best practices. It enables you to implement access control rules to your Kube API; you may specify the permissions that should be granted to users through RBAC.
+
+Although RBAC is enabled by default, starting with Kubernetes 1.6, it should be used appropriately. It would be best if you took appropriate measures to minimize permission duplication and deactivate inactive and unused roles so that you can concentrate only on active components and avoid providing unnecessary permissions to a user.
+
+If you want to specify who has access to the Kubernetes API and what rights they have, RBAC may assist you in doing so. When you activate RBAC, you must also disable the old Attribute-Based Access Control (ABAC) mechanism.
+
+When implementing RBAC, you should prefer namespace-specific rights over cluster-wide permissions. Even while debugging, do not give cluster administrator access. It is safer to provide access only when required.
+
+### **Protect etcd with TLS, Firewall and Encryption**
```

Comment:
This is cool but super hard to use. Any guidance we can give the readers on where to start with this? (also, most cloud providers already fix this right?)

Reply:
Yes, agree. Here are a few references that might help:-

https://searchitoperations.techtarget.com/tutorial/Be-selective-with-Kubernetes-RBAC-permissions
https://docs.giantswarm.io/getting-started/rbac-and-psp/
https://www.cncf.io/blog/2018/08/01/demystifying-rbac-in-kubernetes/
https://jeremievallee.com/2018/05/28/kubernetes-rbac-namespace-user.html
https://docs.openshift.com/container-platform/4.1/authentication/using-rbac.html
https://thenewstack.io/three-realistic-approaches-to-kubernetes-rbac/

Reply:
could you ad a couple of links to the post?